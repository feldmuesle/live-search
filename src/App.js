import React from 'react';
import './App.css';

function App() {
  return <div className="App">Exiting stuff to come...</div>;
}

export default App;
