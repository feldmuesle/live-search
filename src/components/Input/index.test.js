import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import { Input } from '.';
import '../../App.css';

describe('Input renders correctly', () => {
  it('renders correctly with given value', () => {
    const component = renderer.create(
      <Input value="sunshine" name="input" onChange={() => {}} onBlur={() => {}} />
    );

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders disabled input', () => {
    const component = renderer.create(
      <Input
        value="sunshine"
        name="input"
        onChange={() => {}}
        onBlur={() => {}}
        disabled
      />
    );

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

describe('handles events', () => {
  it('calls onChange correctly', () => {
    const handleChange = jest.fn();
    render(<Input value="" name="input" onChange={handleChange} onBlur={() => {}} />);
    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: 'sunshine' } });

    expect(handleChange).toHaveBeenCalled();
  });

  it('calls onBlur correctly', () => {
    const handleBlur = jest.fn();
    render(<Input value="" name="input" onChange={() => {}} onBlur={handleBlur} />);
    const input = screen.getByRole('textbox');
    fireEvent.focus(input);
    fireEvent.blur(input);

    expect(handleBlur).toHaveBeenCalled();
  });
});
