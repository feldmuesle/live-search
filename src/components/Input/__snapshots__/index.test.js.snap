// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly renders correctly with given value 1`] = `
<div
  className="form__group"
>
  <input
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder={null}
    required={false}
    type="text"
    value="sunshine"
  />
</div>
`;

exports[`Input renders correctly renders disabled input 1`] = `
<div
  className="form__group"
>
  <input
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder={null}
    required={false}
    type="text"
    value="sunshine"
  />
</div>
`;
